Boyer-Moore/README.markdown

```diff
@@ -158,40 +158,66 @@ Here's an implementation of the Boyer-Moore-Horspool algorithm:
 ```swift
 extension String {
     func indexOf(pattern: String) -> String.Index? {
-    let patternLength = pattern.characters.count
-    assert(patternLength > 0)
-    assert(patternLength <= self.characters.count)
+        // Cache the length of the search pattern because we're going to
+        // use it a few times and it's expensive to calculate.
+        let patternLength = pattern.characters.count
+        assert(patternLength > 0)
```

Comment:
I feel like we should replace these asserts with `guard patternLength > 0, characters.count < patternLength else { return nil }`. What do you think?